### PR TITLE
bootloader_env.sh wrapper; generic CONFIG_OVERLAY_FS_INDEX; flash-mode-2 fix

### PR DESCRIPTION
**Pull request by mlilien**

### CODE REVIEW

File: recipes-omnect/initrdscripts/omnect-os-initramfs/common-sh [hunk 1]
```diff
@@ -312,13 +313,21 @@ flashmode_efi_grub_handling()
...
+
+        run_cmd_stdout mkdir -p /tmp/boot
+        run_cmd_stdout mount ${dest_dev}${p}1 /tmp/boot
+        run_cmd_stdout efibootmgr --create --disk ${dest_dev} --part 1 --label "omnect_os" --loader \\\\EFI\\\\BOOT\\\\bootx64.efi
+        # same entry with another label ... for debug purposes, when booting after flash-mode-{1,2} fails
+        run_cmd_stdout efibootmgr --create --disk ${dest_dev} --part 1 --label "omnect_os " --loader \\\\EFI\\\\BOOT\\\\bootx64.efi
+        efibootmgr -v > /tmp/boot/EFI/BOOT/efibootmgr_entry
+        run_cmd_stdout umount /tmp/boot
     fi
+    return $ret
```
Comment (JanZachmann) [inline recipes-omnect/initrdscripts/omnect-os-initramfs/common-sh:332]: is this also part of the workaround described in changelog? if so, should we add a comment?

Comment (mlilien) [reply recipes-omnect/initrdscripts/omnect-os-initramfs/common-sh:332]: different issue ... currently i wait to reproduce this, hence the comment abound "debug purpose"

**Review by JanZachmann — APPROVED** [review]

File: recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-2 [hunk 2]
```diff
@@ -143,6 +181,13 @@ run_flash_mode_2() {
         p=""
     fi
 
+    setup_fifo || return 1
+    setup_network eth0 || return 1
+    vfat_grub_dd_workaround "${dest_blk}${p}1"
+    run_bmap_tool || return 1
+    vfat_grub_fsck_workaround
```
Comment (HarryWaschkeit) [inline recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-2:188]: if vfat_grub_xxx are really grub specific, shouldn't they run also only on a grub machine i.e., proteced by 'if [[ " ${machine_features} " =~ " grub " ]]; then ... fi' as in vfat_grub_dd_workaround?
at least everything in vfat_grub_fsck_workaround() is always run opposed to vfat_grub_dd_workaround(), and I would expect a uniform handling in both functions if invocation is done unconditionally.

File: recipes-omnect/initrdscripts/omnect-os-initramfs/common-sh [hunk 3]
```diff
@@ -312,13 +313,21 @@ flashmode_efi_grub_handling()
...
             run_cmd_stdout efibootmgr --bootnum ${i} --delete-bootnum || return 1
         done
-        run_cmd_stdout efibootmgr --create --disk ${dest_dev} --part 1 --label "omnect_os" --loader \\\\EFI\\\\BOOT\\\\bootx64.efi || return 1
+
+        run_cmd_stdout mkdir -p /tmp/boot
+        run_cmd_stdout mount ${dest_dev}${p}1 /tmp/boot
+        run_cmd_stdout efibootmgr --create --disk ${dest_dev} --part 1 --label "omnect_os" --loader \\\\EFI\\\\BOOT\\\\bootx64.efi
+        # same entry with another label ... for debug purposes, when booting after flash-mode-{1,2} fails
+        run_cmd_stdout efibootmgr --create --disk ${dest_dev} --part 1 --label "omnect_os " --loader \\\\EFI\\\\BOOT\\\\bootx64.efi
+        efibootmgr -v > /tmp/boot/EFI/BOOT/efibootmgr_entry
```
Comment (HarryWaschkeit) [inline recipes-omnect/initrdscripts/omnect-os-initramfs/common-sh:329]: out of curiosity: is that for informational/debugging purposes only?

Comment (mlilien) [reply recipes-omnect/initrdscripts/omnect-os-initramfs/common-sh:329]: correct

File: recipes-omnect/bootloader_env/bootloader-env/bootloader_env_u-boot.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+commands=("get","list","set","unset")
+argsc=${#}
+
+function help() {
+    echo "usage:"
+    echo "bootloader_env.sh command key [value]"
+    echo "    command: {get,set,unset}"
```
[the remark this answers is not in the document]
Comment (mlilien) [reply recipes-omnect/bootloader_env/bootloader-env/bootloader_env_u-boot.sh]: fixed

File: recipes-omnect/bootloader_env/bootloader-env/bootloader_env_grub.sh [hunk 5]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+grubenv="/boot/EFI/BOOT/grubenv"
+commands=("get","list","set","unset")
+argsc=${#}
+
+function help() {
+    echo "usage:"
+    echo "bootloader_env.sh command key [value]"
+    echo "    command: {get,set,unset}"
```
[the remark this answers is not in the document]
Comment (mlilien) [reply recipes-omnect/bootloader_env/bootloader-env/bootloader_env_grub.sh]: fixed

File: recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-2 (hunk 2 above)
Comment (mlilien) [reply recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-2]: removed

**Review by HarryWaschkeit — APPROVED** [review]

**Pull request merged**